statistics-dashboard-common/src/main/java/eu/europeana/statistics/dashboard/common/internal/RightsCategory.java

```diff
@@ -1,52 +1,93 @@
+    CC_BY_NC("CC BY-NC", List.of("^https?://creativecommons\\.org/licen[cs]es/by-nc/.*$")),
+    CC_BY_NC_SA("CC BY-NC-SA", List.of("^https?://creativecommons\\.org/licen[cs]es/by-nc-sa/.*$$")),
+    CC_BY_NC_ND("CC BY-NC-ND", List.of("^https?://creativecommons\\.org/licen[cs]es/by-nc-nd/.*$")),
+    NO_COPYRIGHT_NON_COMMERCIAL("No Copyright - Non Commercial Re-Use Only", List.of("^https?://rightsstatements\\.org/(vocab|page)/NoC-NC/.*$",
+            "^https?://www.europeana\\.eu/rights/out-of-copyright-non-commercial/.*$")),
+    NO_COPYRIGHT_OTHER_KNOWN_LEGAL("No Copyright - Other Known Legal Restriction", List.of("^https?://rightsstatements\\.org/(vocab|page)/NoC-OKLR/.*$")),
+    IN_COPYRIGHT_EDUCATIONAL("In Copyright - Educational Use Permitted", List.of("^https?://rightsstatements\\.org/(vocab|page)/InC-EDU/.*$")),
+    IN_COPYRIGHT_EU("In Copyright - EU Orphan Work", List.of("^https?://rightsstatements\\.org/(vocab|page)/InC-OW-EU/.*$",
+            "^https?://www.europeana\\.eu/rights/orphan-work-eu.*$")),
+    IN_COPYRIGHT("In Copyright", List.of("^https?://rightsstatements\\.org/(vocab|page)/InC/.*$",
+            "^https?://www.europeana\\.eu/rights/rr-f/.*$")),
+    COPYRIGHT_NOT_EVALUATED("Copyright Not Evaluated", List.of("^https?://rightsstatements\\.org/(vocab|page)/CNE/.*$",
+            "^https?://www.europeana\\.eu/rights/unknown/.*$")),
+    UNKNOWN("Unknown", List.of());
 
     private final String name;
-    private final String regularExpression;
+    private final List<Pattern> regularExpressions;
 
-    RightsCategory(String name, String regularExpression){
+    RightsCategory(String name, List<String> regularExpressions) {
         this.name = name;
-        this.regularExpression = regularExpression;
+        this.regularExpressions = regularExpressions.isEmpty() ? new ArrayList<>() :
+                regularExpressions.stream().map(Pattern::compile).collect(Collectors.toList());
     }
 
     public String getName() {
         return name;
     }
 
-    public String getRegularExpression() {
-        return regularExpression;
+    public List<Pattern> getRegularExpressions() {
+        return regularExpressions;
     }
 
-    public static RightsCategory matchCategoryFromUrl(String url){
-        Set<RightsCategory> setRightsCategory = Arrays.stream(RightsCategory.values()).filter(x -> x != UNKNOWN).collect(Collectors.toSet());
+    /**
+     * Method that return a category that its regexes match the given url.
+     * Returns Unknown if the url matches none.
+     *
+     * @param url - The url to match to a category
+     * @return The category that its regexes match the given url. Unknown if matches none.
+     */
+    public static RightsCategory matchCategoryFromUrl(String url) {
+        Set<RightsCategory> setRightsCategory = Arrays.stream(RightsCategory.values()).filter(category -> category != UNKNOWN).collect(Collectors.toSet());
```

Comment:
How about

```suggestion
return Arrays.stream(RightsCategory.values()).filter(category -> category != UNKNOWN)
.filter(category -> category.getRegularExpressions().stream()
.anyMatch(regex -> regex.matcher(url).matches()))
.findFirst().orElse(RightsCategory.UNKNOWN);
```

Reply:
I like it! I will update the code to this